storage/azure/src/frontend/AzureFrontendStorage.ts

```diff
@@ -35,22 +32,29 @@ export class AzureFrontendStorage extends FrontendStorage {
     input: (FrontendUrlDownloadInput | FrontendAzureConfigDownloadInput) & {
       transferType: "buffer";
     }
-  ): Promise<Buffer>;
+  ): Promise<ArrayBuffer>;
 
   public download(
     input: (FrontendUrlDownloadInput | FrontendAzureConfigDownloadInput) & {
       transferType: "stream";
     }
-  ): Promise<Readable>;
+  ): Promise<ReadableStream>;
 
   public async download(
     input: FrontendUrlDownloadInput | FrontendAzureConfigDownloadInput
   ): Promise<FrontendTransferData> {
-    const downloadStream = await this._clientWrapperFactory
+    const downloadBlob = await this._clientWrapperFactory
```

Comment:
All `ClientStorage` implementations have some user input validation, what is applicable to `FrontendStorage` is `ObjectReference` validation so that `relativeDirectory` property would not have invalid symbols:
https://github.com/iTwin/object-storage/blob/d1b6c255ee76ce4af6b5b96d6526e676028107a8/storage/azure/src/client/AzureClientStorage.ts#L62

I think we should have similar behavior in `FrontendStorage` as well. This should probably go into a separate PR as the current one is mostly reviewed and the change would affect a few classes.

Reply:
True, let me know if it should be done in a different PR

Reply:
Resolving this for now but yes, this should be a separate PR.